src/track/heaptrack.sh.cmake

```diff
@@ -194,6 +194,15 @@ LIBHEAPTRACK_INJECT=$(readlink -f "$LIBHEAPTRACK_INJECT")
 pipe=/tmp/heaptrack_fifo$$
 mkfifo $pipe
 
+# if say root is profiling a process for non root
+# give actual process access to the pipe
+if [ ! -z "$pid" ]; then
+  PROCESS_USER=$(ps -o user= "$pid" | xargs)
```

Comment:
why do we need the `xargs` call here?

I think I'd prefer this direct approach:

```
if [ ! -z "$pid" ]; then
    pid_user=$(stat -c %U" /proc/"$pid")
    if [ -zu "$pid_user" ]; then
        exit 1;
    fi
    chown "$pid_user" "$pipe" || exit 1;
fi
```
it should be safe to always call this, if the user is the same it doesn't really matter - I don't think we need to optimize